students/JeffBennett/Lesson04/mailroom_2.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+
+import datetime
+import sys
+
+# Data
+
+donors = {
+    'Lorenzo Currie': [3000, 250, 5000],
+    'Marie Kane': [1000, 195, 50.50],
+    'John Beresford Tipton': [10000, 3000, 2500],
+    'Janna Adams': [35.50, 19.75],
+    'LeSean Haskell': [100],
+    'Laila Samir': [1000, 250]
+}
+
+letter_to_all_donors = '''\nDear {}\n\n
+I write to thank you again for all the support you have provided to Northwest
+Lifeboats.  During the past year your total gifts of ${:,.2f} have greatly
+furthered our activities.
+\n\nI have exciting news that one of our prospective donors may be prepared
+to match your next donation, which would double your impact on our efforts.
+I will be back in touch with you soon with more details.\n\nSincerely yours
+\n\nJ A Bennett\nDirector'''
+
+# Input/Output
+
+
+def print_main_menu():
+    """print main menu and prompt for user choice."""
+    print('\nMain Menu\n' + '-'*25 +
+          '\n(1)\tSend a Thank You\n(2)\tCreate a Report'
+          '\n(3)\tSend Letters to All Donors\n(4)\tQuit')
+
+
+def print_donor_list(donors):
+    """print list of donor names."""
+    print('\nOur Current List of Donors:\n' + '-' * 25)
+    for key in donors:
+        print(key)
+
+
+def print_email_to_donor(donor, gift):
+    """print email thanking donor for recent donation."""
+    email_dict = {
+        'greeting': f'\nDear {donor}',
+        'body': f'\n\nThank you very much for your gift of ${gift:,.2f}',
+        'signature': '\n\nSincerely yours\n\nJ A Bennett\nDirector'
+    }
+    email = "{greeting}{body}{signature}".format(**email_dict)
+    print(email)
+
+
+def print_sorted_donors(sorted_donors):
+    """print list of donors sorted by total donation history."""
+    print('{0:32}{1:13}{2:14}{3:13}'.format
+          ('Donor Name', '| Total Gifts', '| Num Gifts', '| Average Gift'))
+    print('-' * 73)
+    for row in sorted_donors:
+        print('{0:30s}  ${1:12.2f}{2:12d}  ${3:12.2f}'.format(*row))
+
+
+# Processing
+
+
+def send_thank_you():
+    """acknowledge gift with an email and update donor and gift list."""
+    while True:
+        donor_name = input("\nEnter donor first and last name,\n " +
+                           "'list' for a list of current donors: \n" +
+                           "  or 'menu' to return to the main menu: ")
+        if donor_name == 'menu':
+            break
+        elif donor_name == 'list':
+            print_donor_list(donors)
+            continue
+        else:
+            gift = float(input("\nEnter amount of gift: " +
+                               " or '0' to return to the main menu: "))
+            if gift == 0:
+                break
+            print_email_to_donor(donor_name, gift)
+            update_donors(donors, donor_name, gift)
+            continue
```

Comment:
you don't need the `continue` here and line 75, it's the last line in your if/else blocks